cpp/marc_tools/single_marc_convert_keywords.cc

```diff
@@ -0,0 +1,156 @@
+/** \brief Utility for comparing keywords with gnd database.
+ *  \author Hjordis Lindeboom
+ *
+ *  \copyright 2021 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <stdexcept>
+#include <utility>
+#include <climits>
+#include <cstdio>
+#include <cstdlib>
+#include "FileUtil.h"
+#include "MARC.h"
+#include "StringUtil.h"
+#include "TextUtil.h"
+#include "util.h"
+
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("gnd_input keyword_input keyword_matches_output keyword_without_matches_output"
+               "       Searches for keyword matches in the \"gnd_input\" file."
+               "       Returns a \"keyword_matches_output\" file with matching keywords and their PPN,"
+               "       as well as \"keywords_without_matches\" file containing keywords where no matches were found.\n");
+}
+
+
+void ExtractSubfieldsForTag(const MARC::Record &record, const std::string &subfield_tag, const std::string &subfield_codes, std::vector<std::string> * const subfields) {
+    const auto gnd_and_fields_iterator(record.findTag(subfield_tag));
+    if (record.findTag(subfield_tag) == record.end())
+        return;
+    const MARC::Subfields subfield_tag_subfields(gnd_and_fields_iterator->getSubfields());
```

Comment:
The name subfield_tag_subfields is wrong and confusing.  Just name it "subfields", please.

Reply:
Also, you could even replace MARC:Subfields with "auto" bit I won't insist on it.